From 42efd6be6cca1a97675353390f1b6cd51e7054df Mon Sep 17 00:00:00 2001
From: Asumu Takikawa <asumu@ccs.neu.edu>
Date: Sat, 2 May 2015 02:15:05 -0400
Subject: [PATCH 2/2] Add docs for unsafe TR operations


diff --git a/typed-racket-doc/typed-racket/scribblings/reference/unsafe.scrbl b/typed-racket-doc/typed-racket/scribblings/reference/unsafe.scrbl
new file mode 100644
index 0000000..1dc7771
--- /dev/null
+++ b/typed-racket-doc/typed-racket/scribblings/reference/unsafe.scrbl
@@ -0,0 +1,30 @@
+#lang scribble/manual
+
+@begin[(require (for-label (only-meta-in 0 [except-in typed/racket for])))]
+
+@title{Unsafe Typed Racket operations}
+
+@defmodule[typed/racket/unsafe]
+
+@bold{Warning}: the operations documented in this section are @emph{unsafe},
+meaning that they can circumvent the invariants of the type system. Unless the
+@racket[#:no-optimize] language option is used, this may result in unpredictable
+behavior and may even crash Typed Racket.
+
+@defform[(require/typed/unsafe m rt-clause ...)]{
+  This form requires identifiers from the module @racket[m] with the same
+  import specifications as @racket[require/typed].
+
+  Unlike @racket[require/typed], this form is unsafe and will not generate
+  contracts that correspond to the specified types to check that the values
+  actually match their types.
+}
+
+@defform[(provide/unsafe provide-spec ...)]{
+  This form declares exports from a module with the same syntax as
+  the @racket[provide] form.
+
+  Unlike @racket[provide], this form is unsafe and Typed Racket will not generate
+  any contracts that correspond to the specified types. This means that uses of the
+  exports in other modules may circumvent the type system's invariants.
+}
diff --git a/typed-racket-doc/typed-racket/scribblings/ts-reference.scrbl b/typed-racket-doc/typed-racket/scribblings/ts-reference.scrbl
index f0f9ca4..e306f5c 100644
--- a/typed-racket-doc/typed-racket/scribblings/ts-reference.scrbl
+++ b/typed-racket-doc/typed-racket/scribblings/ts-reference.scrbl
@@ -34,6 +34,7 @@ For a friendly introduction, see the companion manual
 @include-section["reference/no-check.scrbl"]
 @include-section["reference/typed-regions.scrbl"]
 @include-section["reference/optimization.scrbl"]
+@include-section["reference/unsafe.scrbl"]
 @include-section["reference/legacy.scrbl"]
 @include-section["reference/compatibility-languages.scrbl"]
 @include-section["reference/experimental.scrbl"]
-- 
2.4.6

